src/utils/gitUtils.ts

```diff
@@ -0,0 +1,121 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { MessageItem, Uri } from "vscode";
+import { DialogResponses, IActionContext } from "vscode-azureextensionui";
+import { IStaticWebAppWizardContext } from "../commands/createStaticWebApp/IStaticWebAppWizardContext";
+import { ext } from "../extensionVariables";
+import { getGitApi } from "../getExtensionApi";
+import { API, CommitOptions, Ref, Repository } from "../git";
+import { hasAdminAccessToRepo, tryGetRemote, tryGetReposGetResponseData } from "./gitHubUtils";
+import { localize } from "./localize";
+
+export type GitWorkspaceState = { repo: Repository | null, dirty?: boolean, remoteUrl?: string; hasAdminAccess?: boolean };
+
+export async function getGitWorkspaceState(context: IActionContext & Partial<IStaticWebAppWizardContext>, uri: Uri): Promise<GitWorkspaceState> {
+    const gitWorkspaceState: GitWorkspaceState = { repo: null, dirty: false, remoteUrl: undefined, hasAdminAccess: false };
+    const gitApi: API = await getGitApi();
+    const repo: Repository | null = gitApi.getRepository(uri);
+
+    if (repo) {
+        const originUrl: string | undefined = await tryGetRemote(uri.fsPath);
+        gitWorkspaceState.repo = repo;
+        gitWorkspaceState.dirty = !!repo.state.workingTreeChanges.length;
+        gitWorkspaceState.remoteUrl = originUrl;
+        gitWorkspaceState.hasAdminAccess = originUrl ? hasAdminAccessToRepo(await (tryGetReposGetResponseData(context, originUrl))) : false;
+    }
+
+    return gitWorkspaceState;
+}
+
+export async function verifyRepoForCreation(context: IActionContext, gitWorkspaceState: GitWorkspaceState, uri: Uri): Promise<Repository> {
+    if (!gitWorkspaceState.repo) {
+        const gitRequired: string = localize('gitRequired', 'A GitHub repository is required to create a Static Web App. Would you like to initialize your project as a git repository and create a GitHub remote?');
+        context.telemetry.properties.cancelStep = 'initRepo';
+
+        await ext.ui.showWarningMessage(gitRequired, { modal: true }, DialogResponses.yes);
+        const gitApi: API = await getGitApi();
+        const newRepo: Repository | null = await gitApi.init(uri);
```

Comment:
If this fails, is there a way to get more info about the error? It would be nice to direct that to the output channel for more detail

Reply:
I actually haven't seen this fail yet, but I think it _might_ throw its own error if it does.  I have the check there as more of a fail safe/type safety, but I don't expect that we'll actually see many errors here.